### DISCUSSION TITLE
Export max age of not completed dag to statsd

### DISCUSSION THREAD

**Discussion by volkov** [apache/airflow#25487] [Q&A]
I want to setup monitoring in case of one of dags running too long.

There is  `dagrun.duration.success.<dag_id>` and `dagrun.duration.failed.<dag_id>` metrics, but they are reported only when dag run is finished.

Is there ready metric for this? Can't find it in documentation.
Maybe there is some better solution or workaround for my problem?

The only workaround that I see is to look at `airflow_dagrun_duration_count` and alert if there is no successful tasks for some period of time, it

There is related discussion [24189](https://github.com/apache/airflow/discussions/24189) about task SLA, but looks like there is no easy way to do what I want.

**Comment by potiuk** [comment]
Yeah. contribution and maybe improving SLA to get it improved would be nice.

**Reply by volkov** [reply]
@potiuk SLA redesign is definitely long story.

Is there easier way to add only max age metric?
Can you please tell me where to start developing this metric, if it is good idea?

**Reply by potiuk** [reply]
Well. Just look at the code. I do not know by heart, I would have to dig deeper to point you out but probably better you just start looking at how other metrics are produced and work backwards from there.
